src/Diagnostics/FxCop/System.Runtime.Analyzers/CSharp/Globalization/UseOrdinalStringComparison.Fixer.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace System.Runtime.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class CSharpUseOrdinalStringComparisonFixer : UseOrdinalStringComparisonFixerBase
+    {
+        protected override bool IsInArgumentContext(SyntaxNode node)
+        {
+            return node.IsKind(SyntaxKind.Argument) &&
+                   ((ArgumentSyntax)node).Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression);
+        }
+
+        protected override Task<Document> FixArgument(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode argument)
+        {
+            var memberAccess = ((ArgumentSyntax)argument)?.Expression as MemberAccessExpressionSyntax;
+            if (memberAccess != null)
+            {
+                // preserve the "IgnoreCase" suffix if present
+                bool isIgnoreCase = memberAccess.Name.GetText().ToString().EndsWith(UseOrdinalStringComparisonAnalyzer.IgnoreCaseText, StringComparison.Ordinal);
+                var newOrdinalText = isIgnoreCase ? UseOrdinalStringComparisonAnalyzer.OrdinalIgnoreCaseText : UseOrdinalStringComparisonAnalyzer.OrdinalText;
+                var newIdentifier = syntaxFactoryService.IdentifierName(newOrdinalText);
+                var newMemberAccess = memberAccess.WithName((SimpleNameSyntax)newIdentifier).WithAdditionalAnnotations(Formatter.Annotation);
+                var newRoot = root.ReplaceNode(memberAccess, newMemberAccess);
```

Comment:
Maybe but I think that currently falls in the 20% bucket were you are expected to dive in to the language specific parts.